src/main/java/org/jabref/logic/database/DuplicateCheck.java

```diff
@@ -337,6 +340,16 @@ public boolean isDuplicate(final BibEntry one, final BibEntry two, final BibData
             return false;
         }
 
+        // In case an ISBN is present, it is a strong indicator that the entries are equal.
+        // Only in InBook, InCollection, or Article the ISBN may be equal and the puplication on different pages (and thus not equal)
+        Optional<ISBN> oneISBN = one.getISBN();
+        Optional<ISBN> twoISBN = two.getISBN();
+        if (oneISBN.isPresent() && twoISBN.isPresent()
+                && !List.of(StandardEntryType.Article, StandardEntryType.InBook, StandardEntryType.InCollection).contains(one.getType())
+                && Objects.equals(oneISBN, twoISBN)) {
```

Comment:
I would flip List.of and `Objects.equals`, because `oneISBN` is checked in line 347 and then the subsequent line 348 is closer. Otherwise, the condition is harder to read, because it switches between ISBN, types, and ISBN.